src/themes/default/pages/Category.vue

```diff
@@ -134,21 +143,33 @@ export default {
     },
     getBreadcrumbs () {
       return this.$store.getters['category-next/getBreadcrumbs'].filter(breadcrumb => breadcrumb.name !== this.getCurrentCategory.name)
+    },
+    categoryProducts () {
+      if (isServer) {
+        return ssrProducts
+      }
+
+      return this.getCategoryProducts
     }
   },
   async asyncData ({ store, route }) { // this is for SSR purposes to prefetch data - and it's always executed before parent component methods
     await composeInitialPageState(store, route)
+    ssrProducts = store.getters['category-next/getCategoryProducts']
```

Comment:
I don't like this `ssrProducts` local variable we should probably hide it inside the `category-next/getCategoryProducts` getter; it would be much cleaner and easier to understand if user will be to use only this getter and to have the logick distinguishing if it's CSR (so the state should be returned) or SSR (so the local kept reference should be returned instead)

Reply:
hmm @pkarw right, hiding this inside of getter sound much better but I'm wondering if I'm able to get access to this data... but I'll check, for sure worth to investigate 👍